dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/transport/AbstractEndpoint.java

```diff
@@ -51,7 +51,7 @@ public AbstractEndpoint(URL url, ChannelHandler handler) {
     }
 
     protected static Codec2 getChannelCodec(URL url) {
-        String codecName = url.getParameter(Constants.CODEC_KEY, "telnet");
+        String codecName = url.getProtocol(); // codec extension name must stay the same with protocol name
```

Comment:
what's the purpose for this change?
does this a break change?

Reply:
@chickenlj

Reply:
same question